lib/enmeshed/connector.rb

```diff
@@ -8,119 +8,131 @@ class Connector
+
+        response.status == 200
       end
-      Rails.logger.debug do
-        "Enmeshed::ConnectorApi responded to RelationshipChange with: #{action}; connector response status is #{response.status}"
+
+      # @return [Boolean] Whether the relationship was rejected successfully.
+      def reject_relationship(relationship_id)
+        response = connection.put("/api/v2/Relationships/#{relationship_id}/Reject")
+        Rails.logger.debug do
+          'Enmeshed::ConnectorApi rejected the relationship; ' \
+            "connector response status is #{response.status}"
+        end
+
+        response.status == 200
       end
 
-      response.status == 200
-    end
+      private
+
+      # @return [klass, Array<klass>]
+      # @raise [ConnectorError] If the response contains an error or cannot be parsed.
+      def parse_result(response, klass)
+        json = JSON.parse(response.body).deep_symbolize_keys
 
-    # @return [klass, Array<klass>]
-    # @raise [ConnectorError] If the response contains an error or cannot be parsed.
-    def self.parse_result(response, klass)
-      json = JSON.parse(response.body).deep_symbolize_keys
+        if json.include?(:error)
+          raise ConnectorError.new(
+            "Enmeshed connector response contained error: #{json[:error][:message]}. " \
+            "Full response was: #{response.body}"
+          )
+        end
 
-      if json.include?(:error)
-        raise ConnectorError.new(
-          "Enmeshed connector response contained error: #{json[:error][:message]}. Full response was: #{response.body}"
-        )
+        parse_enmeshed_object(json[:result], klass)
+      rescue JSON::ParserError
+        raise ConnectorError.new("Enmeshed connector response could not be parsed. Received: #{response.body}")
       end
 
-      parse_enmeshed_object(json[:result], klass)
-    rescue JSON::ParserError
-      raise ConnectorError.new("Enmeshed connector response could not be parsed. Received: #{response.body}")
-    end
-    private_class_method :parse_result
-
-    # @return [klass, Array<klass>] The parsed object or array of objects.
-    def self.parse_enmeshed_object(content, klass)
-      if content.is_a?(Array)
-        content.map {|object| klass.parse(object) }
-      else
-        klass.parse(content)
+      # @return [klass, Array<klass>] The parsed object or array of objects.
+      def parse_enmeshed_object(content, klass)
+        if content.is_a?(Array)
+          content.map {|object| klass.parse(object) }
+        else
+          klass.parse(content)
+        end
       end
-    end
-    private_class_method :parse_enmeshed_object
 
-    # @return [Faraday::Connection] The connection to the enmeshed connector.
-    def self.conn
-      @conn ||= init_conn
-    end
-    private_class_method :conn
+      # @return [Faraday::Connection] The connection to the enmeshed connector.
+      # @raise [ConnectorError] If the connector is not configured as expected.
+      def connection
```

Comment:
With this new version, you are no longer memorizing the connection. By creating a new connection object each time, we perform the checks all over again and can't use HTTP keep alive (i.e., as supported by [faraday-net_http_persistent](https://github.com/lostisland/faraday-net_http_persistent)).